### DISCUSSION TITLE
Bin path

### DISCUSSION THREAD

**Discussion by hxhgxy** [Excel-DNA/ExcelDna#640] [Q&A]
![binPath](https://github.com/Excel-DNA/ExcelDna/assets/37101628/61e7521d-09a4-4521-9d6f-5deb8f037cf9)
I use the code above to get the bin folder path, because I put sqlite database there.

It works all along. But error come out suddenly and I found that the code above return the folder of Excel.exe.

How can I can bin folder in class library project (run Excel as startup application)?

**Comment by hxhgxy** [comment]
I just uninstall Microsoft 365 and re-install it. I did nothing with my WPF class library project, the error is not there anymore.

**Comment by hxhgxy** [comment]
Together with the error mentioned above, the Ribbon disappeared.
Is it the possible reason, the Microsoft 365 is upgraded to beta version? And ExcelDNA cannot support the beta version?
(in order to try Python function in Excel, I applied for beta version recently)

**Comment by govert** [comment]
You can get the path to the .xll file by calling `ExcelDnaUtil.XllPath` or `ExcelDnaUtil.XllPathInfo`.

The most common reason why the ribbon does not display is because the Ribbon COM add-in (which Excel-DNA uses behind the scenes) is marked as 'Disabled' by Excel. This can happen after there is an error while Excel runs the ribbon code.
You can check for this with File -> Options -> Add-Ins, then go to the bottom where it says 'Manage:' and select 'Disabled Add-Ins'. Look for your add-in there, and re-enable and the ribbon should work again.

**Comment by hxhgxy** [comment]
Thank you very much!
There is no items under disabled add-ins.

When I run the project, Excel opens without ribbon. And if add the add-in manually via development tools - excel add-ins, I got error message like this:
![ExcelDNA2](https://github.com/Excel-DNA/ExcelDna/assets/37101628/a5196ab5-2e11-45e1-a9e2-4dd209cc2fa8)

**Comment by govert** [comment]
That can happen when you load the add-in a second time. (Maybe the first time it automatically loaded, then you select File->Open which reloads it).

**Comment by hxhgxy** [comment]
I cannot figure out. anyway, I re-installed the microsoft 365.

**Comment by hxhgxy** [comment]
![CTPFocus](https://github.com/Excel-DNA/ExcelDna/assets/37101628/ec9f6ccf-414a-41e5-b696-5b7630b0e2d5)
The searchbar on CTP is focused currently, when I press Enter key, the Excel cell selection move down. It is very strange.

This happens sometimes.

**Comment by govert** [comment]
Maybe the WPF testbox doesn't handle the Enter key by default, so it is passed up to the parent window.
See https://learn.microsoft.com/en-us/dotnet/desktop/wpf/advanced/how-to-detect-when-the-enter-key-pressed
